testutil-server/src/main/java/io/spine/testing/server/blackbox/BlackBox.java

```diff
@@ -673,4 +695,13 @@ public SubscriptionFixture subscribeTo(Topic topic) {
         result.activate();
         return result;
     }
+
+    /**
+     * Returns a factory of {@link io.spine.client.Client Client}s
+     * to this instance of {@code BlackBox}.
+     */
+    public BlackBoxClients clients() {
+        BlackBoxClients result = new BlackBoxClients(this, clientSupplier);
```

Comment:
As discussed, that is because we want to explicitly close the factory of clients in our `BlackBox.close()` method. And having both `blackBoxClients` and `clientSupplier` seems redundant. 

Besides, `BlackBoxClients` consumes little to none resources by itself. It's rather a method object, which just groups and exposes the part of the `BlackBox`'s public API.